File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/H. One Prime/main.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/H
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    unsigned int num;
    bool isPrime = true;
    cin >> num;
    for (int i = 2; i <= sqrt(num); i++)
    {
        if (num % i == 0)
        {
            cout << "NO\n";
            return 0;
        }
    }
    cout << "YES\n";
}

File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/H. One Prime/main.cs
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/H

using System;
using static System.Console;
using static System.Math;

namespace Codeforces.Assuit.Contest1
{
    class ProblemH
    {
        static void Main(string[] args)
        {
            uint num = uint.Parse(Console.ReadLine());

            for (int i = 2; i <= Sqrt(num); i++)
            {
                if (num % i == 0)
                {
                    Write("NO\n");
                    return;
                }
            }
            Write("YES\n");
        }
    }
}
